src/customRules/importTypeOrderRule.ts

```diff
@@ -0,0 +1,66 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+
+export class Rule extends Lint.Rules.AbstractRule {
+  /* tslint:disable:object-literal-sort-keys */
+  public static metadata: Lint.IRuleMetadata = {
+    'ruleName': 'import-type-order',
+    'description': 'Enforce structure to your imports. Import structure should be listed in the following order: modules, absolute imports,  relative parent/ancestor directories, relative sibling directors.',
+    'hasFix': false,
+    'optionsDescription': 'Not configurable.',
+    'options': null,
+    'optionExamples': null,
+    'type': 'style',
+    'typescriptOnly': true,
+  };
+  /* tslint:enable:object-literal-sort-keys */
+
+  public static IMPORT_TYPE_ORDER_ERROR = 'Imports should be listed in the following order: module imports, absolute imports, ancestor imports, sibling imports.';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    return this.applyWithFunction(sourceFile, walker);
+  }
+}
+
+enum ImportType {
+  External = 0, // 'external'
+  Absolute = 1, // '/some/folder/file'
+  Ancestor = 2, // '../parentFolder'
```

Comment:
I'm still on the fence about `Ancestor`, but I can't think of anything better.  `Internal`?  `Local`?  No.  Bah.

Ohwell, they're not used outside this file, so we can easily change when inspiration strikes 😄